File: client_app/android/app/src/main/java/com/aionsigma_app/LocationUtils.java
package com.aionsigma_app;


import android.Manifest;
import android.content.Context;
import android.content.pm.PackageManager;
import android.location.Location;
import android.location.LocationManager;
import android.support.v4.app.ActivityCompat;

public class LocationUtils {
    Context context;
    private static final int REQUEST_LOCATION = 1;

    public LocationUtils(Context context){
        this.context= context;
    }

    public Location getLocation() {
        if (ActivityCompat.checkSelfPermission(context, Manifest.permission.ACCESS_FINE_LOCATION)
                != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission
                (context, Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED) {

            //ActivityCompat.requestPermissions(context, new String[]{Manifest.permission.ACCESS_FINE_LOCATION}, REQUEST_LOCATION);
            return null;
        } else {
            String lattitude,longitude;
            LocationManager  locationManager = (LocationManager) context.getSystemService(Context.LOCATION_SERVICE);

            Location location = locationManager.getLastKnownLocation(LocationManager.NETWORK_PROVIDER);
            Location location1 = locationManager.getLastKnownLocation(LocationManager.GPS_PROVIDER);
            Location location2 = locationManager.getLastKnownLocation(LocationManager. PASSIVE_PROVIDER);

            if (location != null) {
                return location;
            } else  if (location1 != null) {
                return location1;


            } else  if (location2 != null) {
                /*double latti = location2.getLatitude();
                double longi = location2.getLongitude();
                lattitude = String.valueOf(latti);
                longitude = String.valueOf(longi);
                */
                return location2;
            }else{
                return null;
            }
        }
    }
}

File: android/AionSigma/app/src/main/java/com/aionsigma/android/Ultilities/LocationUtils.kt
package com.aionsigma.android.Ultilities

import android.Manifest
import android.content.Context
import android.content.pm.PackageManager
import android.location.Location
import android.location.LocationManager
import android.support.v4.app.ActivityCompat

class LocationUtils(internal var context: Context) {

    //ActivityCompat.requestPermissions(context, new String[]{Manifest.permission.ACCESS_FINE_LOCATION}, REQUEST_LOCATION);
    /*double latti = location2.getLatitude();
                double longi = location2.getLongitude();
                lattitude = String.valueOf(latti);
                longitude = String.valueOf(longi);
                */ val location: Location?
        get() {
            if (ActivityCompat.checkSelfPermission(context, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(context, Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
                return null
            } else {
//                val lattitude: String
//                val longitude: String
                val locationManager = context.getSystemService(Context.LOCATION_SERVICE) as LocationManager

                val location = locationManager.getLastKnownLocation(LocationManager.NETWORK_PROVIDER)
                val location1 = locationManager.getLastKnownLocation(LocationManager.GPS_PROVIDER)
                val location2 = locationManager.getLastKnownLocation(LocationManager.PASSIVE_PROVIDER)

                return location ?: (location1 ?: location2)
            }
        }

    companion object {
        private val REQUEST_LOCATION = 1
    }
}